crates/common/src/state_update.rs

```diff
@@ -208,6 +210,186 @@ impl StateUpdate {
                 })
         })
     }
+
+    /// Compute the state diff commitment used in block commitment signatures.
+    ///
+    /// How to compute the value is documented in [this Starknet Community article](https://community.starknet.io/t/introducing-p2p-authentication-and-mismatch-resolution-in-v0-12-2/97993).
+    pub fn state_diff_commitment(&self) -> StateDiffCommitment {
```

Comment:
Should we communicate/emphasize that these are expensive and not just getters? Genuine question, I'm unsure. Ditto for the private functions.
```suggestion
    pub fn compute_state_diff_commitment(&self) -> StateDiffCommitment {
```

Reply:
OK, I like the idea. `compute_state_diff_commitment()` is a better name for this function.

I've also renamed the private methods -- I don't think it matter that much there since those are not exposed anywhere but `compute_` is more precise.